root@mailgate2:/etc/mail# rcsdiff -u -r1.1 ../rc.d/rc.scamback
===================================================================
RCS file: ../rc.d/RCS/rc.scamback,v
retrieving revision 1.1
diff -u -r1.1 ../rc.d/rc.scamback
--- ../rc.d/rc.scamback 2020/05/02 12:14:04     1.1
+++ ../rc.d/rc.scamback 2020/05/14 12:34:58
@@ -1,6 +1,7 @@
 #!/bin/sh
 # Start/stop/restart scamback.

+RETVAL=0
 # Start scam-back:
 scamback_start() {
   # Check for dead socket existence
@@ -57,6 +58,33 @@

 # Status scam-back:
 scamback_status() {
+  # Check for dead socket existence
+  if [ -S /var/spool/scamback/scam-back.sock ] ; then
+    # check to see if scam-back is already running
+    # Socket exists
+    spid=$(fuser /var/spool/scamback/scam-back.sock 2>/dev/null | tr -d ' ')
+    if [ -f /var/spool/scamback/scam-back.pid ] ; then
+      # set pid to look for
+      pid=`cat /var/spool/scamback/scam-back.pid`
+    else
+      # check process table for scam-back
+      pid=`ps auxw | grep -v grep | grep scam-back | grep '^scamback' | awk '{print $2}'`
+    fi
+    if [ "X${pid}" = "X" -o "X${spid}" = "X" ] ; then
+      # pid does not seem to be set
+      # and socket is not open with a pid
+      # we can blow away socket safely
+      rm -f /var/spool/scamback/scam-back.sock
+      echo "scamback: scamback is not running."
+    fi
+    test=`ps auxw | grep -v grep | grep "${pid} " | grep scam-back | grep -c '^scamback'`
+    # if test greater than 1, then scam-back likely running
+    if [ ${test} -eq 0 ] ; then
+      # we can blow away socket safely
+      rm -f /var/spool/scamback/scam-back.sock
+      echo "scamback: scamback is not running."
+    fi
+  fi
   pspid=`ps auxw | grep -v grep | grep scam-back | grep '^scamback' | awk '{print $2}'`
   if [ ! -f /var/spool/scamback/scam-back.pid ] ; then
     if [ "X${pspid}" != "X" ] ; then
@@ -101,6 +129,7 @@

 case "$1" in
 'start')
+  scamback_status
   scamback_start
   ;;
 'stop')
@@ -108,10 +137,15 @@
   ;;
 'status')
   scamback_status
+  RETVAL=$?
   ;;
 'restart')
   scamback_restart
+  RETVAL=$?
   ;;
 *)
   echo "usage $0 start|stop|status|restart"
+  exit 1
 esac
+
+exit $RETVAL
